src/graph/graphQL.ts

```diff
@@ -116,40 +292,111 @@ export function resolveAndReadFileSync(path: string,
     }
 }
 
-function replaceParameters(query: string,
+function replaceParameters(q: string,
                            parameters: {
                                 [name: string]: string | boolean | number | ParameterEnum;
                             } = {}): string {
     if (Object.keys(parameters).length > 0) {
         const exp = OperationParameterExpression;
-        if (exp.test(query)) {
-            const result = exp.exec(query);
+        if (exp.test(q)) {
+            const result = exp.exec(q);
             // First delete the parameter declaration at the top of the subscription
-            query = query.replace(result[2], "");
+            q = q.replace(result[2], "");
             for (const key in parameters) {
                 if (parameters.hasOwnProperty(key)) {
                     const value = parameters[key] as any;
                     // If value is defined it is a enum value
                     if (value.value) {
                         if (Array.isArray(value.value)) {
-                            query = replace(query, `\\$${key}`, `[${value.value.join(", ")}]`);
+                            q = replace(q, `\\$${key}`, `[${value.value.join(", ")}]`);
                         } else {
-                            query = replace(query, `\\$${key}`, value.value);
+                            q = replace(q, `\\$${key}`, value.value);
                         }
                     } else {
-                        query = replace(query, `\\$${key}`, JSON.stringify(value));
+                        q = replace(q, `\\$${key}`, JSON.stringify(value));
                     }
                 }
             }
 
             // Calulate hash to suffix the subscriptionName
-            const hash = murmur3(query, 37);
-            query = replaceOperationName(query, `${result[1]}_${hash}`);
+            const hash = murmur3(q, 37);
+            q = replaceOperationName(q, `${result[1]}_${hash}`);
         }
     }
-    return query;
+    return q;
 }
 
-function replace(query: string, key: string, value: string): string {
-    return query.replace(new RegExp(`${key}\\b`, "g"), value);
+function replace(q: string, key: string, value: string): string {
+    return q.replace(new RegExp(`${key}\\b`, "g"), value);
+}
+
+function inlineFragments(q: string, name: string, moduleDir: string, fragmentDir: string): string {
```

Comment:
which of these are required? i am going to try to get this too

Reply:
What do you mean? They are all required parameters.